cmd/clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -130,11 +130,18 @@ func (d *ClusterDeployer) Create(cluster *clusterv1.Cluster, machines []*cluster
 
 	// For some reason, endpoint doesn't get updated in bootstrap cluster sometimes. So we
 	// update the target cluster endpoint as well to be sure.
-	klog.Infof("Updating target cluster object with control plane endpoint running on %s", controlPlaneMachine.Name)
+	klog.Infof("Updating target cluster object with control plane endpoint running on %s", controlPlaneMachines[0].Name)
 	if err := d.updateClusterEndpoint(targetClient, provider, cluster.Name, cluster.Namespace); err != nil {
 		return errors.Wrap(err, "unable to update target cluster endpoint")
 	}
 
+	if len(controlPlaneMachines) > 1 {
+		klog.Info("Creating additional controlplane machines in target cluster.")
+		if err := phases.ApplyMachines(targetClient, cluster.Namespace, controlPlaneMachines[1:]); err != nil {
```

Comment:
This should really be done serially until https://github.com/kubernetes/kubeadm/issues/1097 is resolved and all supported versions of k8s we are deploying (using kubeadm) have the fix.

Reply:
Thank you @detiber it now looks like this:
```
I0404 16:54:01.057320   44009 clusterdeployer.go:141] Creating additional control plane machines in target cluster.
I0404 16:54:01.181076   44009 applymachines.go:36] Creating machines in namespace "mv1a"
I0404 16:54:01.299471   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:54:11.430933   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:54:21.433699   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:54:31.432906   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:54:41.433499   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:54:51.433788   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:55:01.433685   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:55:11.434382   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:55:21.433491   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-1 to become ready...
I0404 16:55:21.674655   44009 applymachines.go:36] Creating machines in namespace "mv1a"
I0404 16:55:21.794886   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:55:31.921803   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:56:11.918182   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:56:21.918373   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:56:31.919049   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:56:41.919050   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:56:51.921335   44009 clusterclient.go:972] Waiting for Machine mv1a-controlplane-2 to become ready...
I0404 16:56:52.037719   44009 clusterdeployer.go:149] Creating node machines in target cluster.
I0404 16:56:52.156203   44009 applymachines.go:36] Creating machines in namespace "mv1a"
I0404 16:56:52.156226   44009 clusterdeployer.go:154] Done provisioning cluster. You can now access your cluster with kubectl --kubeconfig kubeconfig
I0404 16:56:52.156668   44009 createbootstrapcluster.go:36] Cleaning up bootstrap cluster.
I0404 16:56:52.156688   44009 kind.go:69] Running: kind [delete cluster --name=mv1a]
I0404 16:56:54.173234   44009 kind.go:72] Ran: kind [delete cluster --name=mv1a] Output: Deleting cluster "mv1a" ...
```